site/app/libraries/database/AbstractDatabase.php

```diff
@@ -158,31 +144,26 @@ public function query($query, $parameters = []) {
                     }
                 }
             }
-            $this->all_queries[] = [$query, $parameters];
-            $statement = $this->link->prepare($query);
-            $result = $statement->execute($parameters);
+            $statement = $this->conn->prepare($query);
+            $result = $statement->executeQuery($parameters);
 
             $this->row_count = null;
             $identity = QueryIdentifier::identify($query);
             if (
                 in_array($identity, [QueryIdentifier::UPDATE, QueryIdentifier::DELETE, QueryIdentifier::INSERT])
             ) {
-                $this->row_count = $statement->rowCount();
+                $this->row_count = $result->rowCount();
             }
             elseif ($identity === QueryIdentifier::SELECT) {
-                $columns = $this->getColumnData($statement);
-                $this->results = $statement->fetchAll(\PDO::FETCH_ASSOC);
-                foreach ($this->results as $idx => $result) {
-                    $this->results[$idx] = $this->transformResult($result, $columns);
-                }
+                $this->results = $result->fetchAllAssociative();
                 $this->row_count = count($this->results);
             }
         }
-        catch (\PDOException $pdoException) {
-            throw new DatabaseException($pdoException->getMessage(), $query, $parameters);
+        catch (DBALException $dbalException) {
+            throw new DatabaseException($dbalException->getMessage(), $query, $parameters);
         }
 
-        return $result;
+        return true;
```

Comment:
Based on the code before, it appears like it would never return false due to it having the ERRMODE_EXCEPTION mode on. This means the PDO execute won't ever return false and will just throw an exception instead. I have only seen 2 places in the codebase that use this return value and they use it as bools. Maybe we should remove the return value entirely since if it fails, it will throw an exception?

Reply:
Discussed in person.  Changes still necessary.